vmdb/app/models/service_orchestration/option_converter.rb

```diff
@@ -38,9 +38,13 @@ def stack_create_options
     end
 
     # factory method to instantiate a provider dependent converter
-    def self.get_converter(dialog_options, manager_class_name)
-      class_name = "ServiceOrchestration::OptionConverter#{manager_class_name.match(/Ems(.*)/)[1]}"
-      class_name.constantize.new(dialog_options)
+    def self.get_converter(dialog_options, manager_class)
+      if manager_class.parent == Object
+        klass = "ServiceOrchestration::OptionConverter#{manager_class.name.match(/Ems(.*)/)[1]}".constantize
+      else
+        klass = manager_class::ServiceOptionConverter
```

Comment:
Because `OptionConverter` is already under namespace `ServiceOrchestration`, there is no confusion. `ServiceOrchestration::ServiceOrchestrationOptionConverter` looks redundant. If our naming convention requires so, I would vote `ServiceOrchestrationOptionConverter`